### PR TITLE
Remove need of proguard config for DCE pass #2

### PR DESCRIPTION
**Pull request by parind**

### CODE REVIEW

File: cmake_modules/Commons.cmake [hunk 1]
```diff
@@ -20,11 +20,11 @@ macro(set_common_cxx_flags_for_redex)
...
 
-        set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -g -Wall")
+        set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -g -Wall -pthread")
         set(COMMON_CXX_FLAGS_NODBG, "-O3")
-        set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} ${COMMON_CXX_FLAGS_NODBG}")
-        set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} ${COMMON_CXX_FLAGS_NODBG}")
-        set(CMAKE_CXX_FLAGS_MINSIZEREL "${CMAKE_CXX_FLAGS_MINSIZEREL} ${COMMON_CXX_FLAGS_NODBG}")
+        set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} ${COMMON_CXX_FLAGS_NODBG} -pthread")
+        set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} ${COMMON_CXX_FLAGS_NODBG} -pthread")
+        set(CMAKE_CXX_FLAGS_MINSIZEREL "${CMAKE_CXX_FLAGS_MINSIZEREL} ${COMMON_CXX_FLAGS_NODBG} -pthread")
```
Comment (justinjhendrick) [inline cmake_modules/Commons.cmake]: nit: why repeat this three times when we could put it on `COMMON_CXX_FLAGS_NODBG`once?

Comment (parind) [reply cmake_modules/Commons.cmake]: It still gave me the Missing DSO error.

Comment (parind) [reply cmake_modules/Commons.cmake]: This was what I did
`set(COMMON_CXX_FLAGS_NODBG, "-O3 -pthread")`
but still I get the following error
`libpthread.so.0: error adding symbols: DSO missing from command line`

Comment (justinjhendrick) [reply cmake_modules/Commons.cmake]: Huh. Okay, we'll keep it like this then because it works. Thanks for the quick response.

**Comment by justinjhendrick** [comment]
Looks good to me. I'll import it and it'll get reviewed by my teammates, then should land some time soon.

**Pull request closed without merging**